extension/firefox/chrome/ShumwayCom.jsm

```diff
@@ -0,0 +1,701 @@
+    Components.utils.exportFunction(getSettings, shumwayComAdapter, {defineAs: 'getSettings'});
+    Components.utils.exportFunction(getPluginParams, shumwayComAdapter, {defineAs: 'getPluginParams'});
+    Components.utils.exportFunction(reportIssue, shumwayComAdapter, {defineAs: 'reportIssue'});
+    Components.utils.exportFunction(externalCom, shumwayComAdapter, {defineAs: 'externalCom'});
+    Components.utils.exportFunction(loadFile, shumwayComAdapter, {defineAs: 'loadFile'});
+    Components.utils.exportFunction(reportTelemetry, shumwayComAdapter, {defineAs: 'reportTelemetry'});
+    Components.utils.exportFunction(setClipboard, shumwayComAdapter, {defineAs: 'setClipboard'});
+    Components.utils.exportFunction(navigateTo, shumwayComAdapter, {defineAs: 'navigateTo'});
+    Components.utils.exportFunction(userInput, shumwayComAdapter, {defineAs: 'userInput'});
+
+    Object.defineProperties(shumwayComAdapter, {
+      onLoadFileCallback: { value: null, writable: true },
+      onExternalCallback: { value: null, writable: true },
+    });
+
+    // Exposing createSpecialInflate function for DEFLATE stream decoding using
+    // Gecko API.
+    if (SpecialInflateUtils.isSpecialInflateEnabled) {
+      Components.utils.exportFunction(function () {
+        return SpecialInflateUtils.createWrappedSpecialInflate(content);
+      }, shumwayComAdapter, {defineAs: 'createSpecialInflate'});
+    }
+
+    if (RtmpUtils.isRtmpEnabled) {
+      Components.utils.exportFunction(function (params) {
+        return RtmpUtils.createSocket(content, params);
+      }, shumwayComAdapter, {defineAs: 'createRtmpSocket'});
+      Components.utils.exportFunction(function () {
+        return RtmpUtils.createXHR(content);
+      }, shumwayComAdapter, {defineAs: 'createRtmpXHR'});
+    }
+
+    Components.utils.makeObjectPropsNormal(shumwayComAdapter);
+
+    return shumwayComAdapter;
+  },
+
+  createActions: function (startupInfo, window, document) {
+    return new ShumwayChromeActions(startupInfo, window, document);
+  }
+};
+
+
+// All the priviledged actions.
+function ShumwayChromeActions(startupInfo, window, document) {
+  this.url = startupInfo.url;
+  this.objectParams = startupInfo.objectParams;
+  this.movieParams = startupInfo.movieParams;
+  this.baseUrl = startupInfo.baseUrl;
+  this.isOverlay = startupInfo.isOverlay;
+  this.embedTag = startupInfo.embedTag;
+  this.isPausedAtStart = startupInfo.isPausedAtStart;
+  this.window = window;
+  this.document = document;
+  this.externalComInitialized = false;
+  this.allowScriptAccess = startupInfo.allowScriptAccess;
+  this.lastUserInput = 0;
+  this.crossdomainRequestsCache = Object.create(null);
+  this.telemetry = {
+    startTime: Date.now(),
+    features: [],
+    errors: []
+  };
+
+  this.onLoadFileCallback = null;
+  this.onExternalCallback = null;
+}
+
+ShumwayChromeActions.prototype = {
+  invoke: function (name, args) {
```

Comment:
At least this function should have a comment explaining what it's for and why it's safe.